model/versioning/Versioned.php

```diff
@@ -301,12 +308,13 @@ public function augmentSQL(SQLSelect $query, DataQuery $dataQuery = null) {
 		// Reading a specific stage (Stage or Live)
 		case 'stage':
 			$stage = $dataQuery->getQueryParam('Versioned.stage');
-			if($stage && ($stage != $this->defaultStage)) {
+			if($this->hasVersionedStage($stage) && $stage !== static::DRAFT_STAGE) {
```

Comment:
The current stage (set via session) might not match the given stage for a particular object; In which case, it should revert to its default stage.

The particular case is where an object only has ONE stage (e.g. `Stage`), but should still be visible if the site is set to set_stage('Live')`. Single-stage objects are a valid use case where we want only the versioning component, but not the staging one.